packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -137,22 +155,40 @@ export class GutenbergEditorPage {
 	}
 
 	/**
-	 * Return the editor iframe.
+	 * Return the editor frame. Could be the top-level frame (i.e WPAdmin).
+	 * an iframe (Calypso/Gutenframe).
 	 *
-	 * @returns {Promise<Frame>} iframe holding the editor.
+	 * @returns {Promise<Frame>} frame holding the editor.
 	 */
 	async getEditorFrame(): Promise< Frame > {
-		const locator = this.page.locator( selectors.editorFrame );
-
-		const elementHandle = await locator.elementHandle( {
-			timeout: 105 * 1000,
-		} );
-
-		if ( ! elementHandle ) {
-			throw new Error( 'Could not locate editor iframe.' );
+		if ( ! this.requiresGutenframe ) {
+			// If we're not in the Gutenframe context (i.e not Calypso), then
+			// just return the top frame, no need to try to locate the iframe.
+			if ( ! this.page.url().startsWith( 'https://wordpress.com' ) ) {
```

Comment:
Ack, it looks like this won't work on PR builds. because of the Calypso live URL! This is why the PR e2e builds failed on this PR.

Hows about searching for `wp-admin` in the URL instead?

Reply:
Ah, good point! I'll do that 👍🏻